Lib/fontTools/subset/__init__.py

```diff
@@ -2258,8 +2258,23 @@ def collect_colors_by_index(paint):
 	self.numPaletteEntries = len(self.palettes[0])
 
 	if self.version == 1:
-		self.paletteEntryLabels = [
-			label for i, label in enumerate(self.paletteEntryLabels) if i in retained_palette_indices
+		kept_labels = []
+		dropped_labels = []
+		for i, label in enumerate(self.paletteEntryLabels):
+			if i in retained_palette_indices:
+				kept_labels.append(label)
+			else:
+				dropped_labels.append(label)
+		self.paletteEntryLabels = kept_labels
+		# Remove dropped labels from the name table.
+		name_table = font["name"]
+		name_table.names = [
```

Comment:
with my black hat on, I think the `or` should be placed on the next line, one condition per line

```python
name_table.names = [
    n
    for n in name_table.names
    if (
        n.nameID not in dropped_labels
        # Only remove nameIDs in the user range and if they're not explicitly kept
        or n.nameID < 256
        or n.nameID in options.name_IDs
    )
]
```